src/game/g_cmds.cpp

```diff
@@ -1808,6 +1808,58 @@ void G_SetClientWeapons(gentity_t *ent, weapon_t w1, weapon_t w2, qboolean updat
 	}
 }
 
+/*
+=================
+G_GetDefaultWeaponForClass
+Returns default weapons for class
+=================
+*/
+int G_GetDefaultWeaponForClass(gentity_t* ent, char* s, int weapon, bool primary, bool updateclient)
```

Comment:
This function could just return `weapon_t` so you don't need a static cast on the caller side.

`weapon`-parameter should be removed as well as it's never used, and this function is only called when we know weapon is `0`.

Suggested signature:
```c
weapon_t G_GetDefaultWeaponForClass(gentity_t* ent, char* s, bool primary, bool updateclient)
```